src/zcl_aff_writer_xslt.clas.abap

```diff
@@ -765,4 +787,30 @@ CLASS zcl_aff_writer_xslt IMPLEMENTATION.
 
   ENDMETHOD.
 
+
+  METHOD write_enum_map_ext_compatible.
+    write_open_tag( line = |<tt:deserialize>| ).
+    write_tag( line = |<tt:read type="C" var="VARIABLE"/>| ).
+
+    DATA index TYPE i.
+    LOOP AT enum_values ASSIGNING FIELD-SYMBOL(<enum_value>).
+      DATA(abap_value) = get_abap_value( abap_value = <enum_value>-abap_value element_description = element_description ).
+      IF <enum_value>-overwritten_json_value IS INITIAL.
+        DATA(xml_value) = <enum_value>-json_value.
+      ELSE.
+        xml_value = <enum_value>-overwritten_json_value.
+      ENDIF.
+      IF index < lines( enum_values ).
+        write_open_tag( |<tt:cond-var check="VARIABLE='{ xml_value }'">| ).
+        write_tag( |<tt:assign { get_to_ref( element_name ) } val="{ abap_value }"/>| ).
+        write_closing_tag( `</tt:cond-var>` ).
+      ELSE.
+        write_open_tag( |<tt:cond-var check="VARIABLE={ xml_value }| ).
+        write_tag( `"/>` ).
```

Comment:
I think we can delete this whole index handling here. So just keep the code in the if-part and remove the variable index